src/Tools/VMC/EventLibraryInterface.cxx

```diff
@@ -0,0 +1,261 @@
+//____________________________________________________________________________
+/*
+ Copyright (c) 2003-2020, The GENIE Collaboration
+ For the full text of the license visit http://copyright.genie-mc.org
+
+
+*/
+//____________________________________________________________________________
+
+#include "Framework/Messenger/Messenger.h"
+#include "Framework/Numerical/RandomGen.h"
+#include "Framework/GHEP/GHepRecord.h"
+#include "Framework/GHEP/GHepParticle.h"
+#include "Framework/ParticleData/PDGUtils.h"
+#include "Framework/ParticleData/PDGLibrary.h"
+#include "Framework/Interaction/Interaction.h"
+#include "Tools/VMC/EventLibraryInterface.h"
+#include "Tools/VMC/RecordList.h"
+#include "Tools/VMC/Utils.h"
+
+#include "TFile.h"
+
+using namespace genie;
+using namespace genie::vmc;
+
+//____________________________________________________________________________
+EventLibraryInterface::EventLibraryInterface() :
+  EventRecordVisitorI("genie::vmc::EventLibraryInterface"),
+  fRecordFile(0)
+{
+
+}
+//____________________________________________________________________________
+EventLibraryInterface::EventLibraryInterface(string config) :
+  EventRecordVisitorI("genie::vmc::EventLibraryInterface", config),
+  fRecordFile(0)
+{
+
+}
+//____________________________________________________________________________
+EventLibraryInterface::~EventLibraryInterface()
+{
+  for(auto it: fRecords) delete it.second;
+  delete fRecordFile;
+}
+//____________________________________________________________________________
+void EventLibraryInterface::ProcessEventRecord(GHepRecord * event) const
+{
+// Get event summary constructed by GENIE
+//
+  Interaction* interaction = event->Summary();
+  const InitialState & init_state = interaction->InitState();
+
+  const Record* rec = GetRecord(interaction);
```

Comment:
The use of generic class name `Record` here within `ProcessEventRecord(GHepRecord * event)` had me do a double-take.  Could it be changed up to something like `EvtLibRecord` ?

Reply:
Can do (or we could be more explicit with namespaces?)

Reply:
Done